src/Components/Feed.tsx

```diff
@@ -82,33 +90,105 @@ const Feed = () => {
   const toggleLike = (index: number) => {
     setLikedPosts(likedPosts.map((liked, i) => (i === index ? !liked : liked)));
   };
-  const Comment = ({ username, text }: { username: string; text: string }) => (
-    <Box mb="xs" style={{ borderBottom: "1px solid black" }}>
+  const toggleComments = (index: number) => {
+    setCommentsVisible(
+      commentsVisible.map((visible, i) => (i === index ? !visible : visible))
+    );
+  };
+  const addComment = (index: number, commentText: string) => {
+    if (commentText.trim()) {
+      const newComment = {
+        username: "User",
+        text: commentText,
+        date: getFormattedDate(),
+      };
+      setComments(
+        comments.map((postComments, i) =>
+          i === index ? [...postComments, newComment] : postComments
+        )
+      );
+      setCommentCounts(
+        commentCounts.map((count, i) => (i === index ? count + 1 : count))
+      );
+    }
+  };
+  const addCommentNumber = (index: number) => {
+    let commentNo = index + 1;
+    return commentNo;
+  };
+  const CommentNumber = ({ commentNo }: { commentNo: number }) => (
+    <Box component="span">{commentNo} comments</Box>
+  );
+
+  const Comment = ({
```

Comment:
Move these components to their own seperate files. You can add it to your components folder

Reply:
okay